File: 필수문항/122_경비원.py
import sys

input = sys.stdin.readline

xl, yl = map(int, input().split())

n = int(input())
stores = [list(map(int, input().split())) for _ in range(n)]

d, l = map(int, input().split())
ans = 0

for i in range(n):
    sd, sl = stores[i]
    if d == 1:
        if sd == 1:
            ans += abs(l - sl)
        elif sd == 2:
            if l + sl <= xl * 2 - l - sl:
                ans += yl + l + sl
            else:
                ans += yl + 2 * xl - l - sl
        elif sd == 3:
            ans += l + sl
        else:
            ans += xl - l + sl
    elif d == 2:
        if sd == 2:
            ans += abs(l - sl)
        elif sd == 1:
            if l + sl <= xl * 2 - l - sl:
                ans += yl + l + sl
            else:
                ans += yl + 2 * xl - l - sl
        elif sd == 3:
            ans += l + yl - sl
        else:
            ans += xl - l + yl - sl
    elif d == 3:
        if sd == 1:
            ans += l + sl
        elif sd == 2:
            ans += yl - l + sl
        elif sd == 3:
            ans += abs(l - sl)
        else:
            if l + sl <= 2 * yl - l - sl:
                ans += l + sl + xl
            else:
                ans += 2 * yl - l - sl + xl
    else:
        if sd == 1:
            ans += l + xl - sl
        elif sd == 2:
            ans += yl - l + xl - sl
        elif sd == 4:
            ans += abs(l - sl)
        else:
            if l + sl <= 2 * yl - l - sl:
                ans += l + sl + xl
            else:
                ans += 2 * yl - l - sl + xl

print(ans)

File: 필수문항/122_경비원.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int x = Integer.parseInt(st.nextToken());
        int y = Integer.parseInt(st.nextToken());
        int n = Integer.parseInt(br.readLine());
        int[] len = new int[n + 1];
        for (int i = 0; i <= n; i++) {
            st = new StringTokenizer(br.readLine());
            int d = Integer.parseInt(st.nextToken());
            int l = Integer.parseInt(st.nextToken());
            switch (d) {
                case 1:
                    len[i] = -y - l;
                    break;
                case 2:
                    len[i] = l;
                    break;
                case 3:
                    len[i] = -y + l;
                    break;
                case 4:
                    len[i] = x + y - l;
                    break;
            }
        }
        int ans = 0;
        int total = 2 * (x + y);
        for (int i = 0; i < n; i++) {
            int l = Math.abs(len[n] - len[i]);
            ans += Math.min(l, total - l);
        }
        System.out.println(ans);
    }
}
